Lib/test/test_asyncio/test_windows_events.py

```diff
@@ -250,6 +250,38 @@ def test_address_argument_type_error(self):
             proactor.sendto(sock, b'abc', addr=bad_address)
         sock.close()
 
+    def test_client_pipe_stat(self):
+        res = self.loop.run_until_complete(self._test_client_pipe_stat())
+        self.assertEqual(res, 'done')
+
+    async def _test_client_pipe_stat(self):
+        ADDRESS = r'\\.\pipe\test_client_pipe_stat-%s' % os.getpid()
+
+        with self.assertRaises(FileNotFoundError):
+            os.stat(ADDRESS)
+
+        [server] = await self.loop.start_serving_pipe(
+            asyncio.Protocol, ADDRESS)
+        self.assertIsInstance(server, windows_events.PipeServer)
+
+        errors = []
+        self.loop.set_exception_handler(lambda *args: errors.append(args))
+
+        async def stat_it():
+            os.stat(ADDRESS)
```

Comment:
In case the implementation of `os.stat()` changes, what basically needs to be tested is whether a pipe connection that's closed immediately causes the server to die, such as `os.close(os.open(ADDRESS, 0))`.

Reply:
I tried it and it does not reproduces the error. Note that I am using Windows 11.

Reply:
Yeah, I found the same problem. I also had to add two `await asyncio.sleep(0)` calls after `server.close()` to make the final `close(open())` call fail as expected.

So I think I'd rather merge this as is, using `os.stat()`.

Reply:
All that `os.stat()` is doing that's even remotely relevant here is calling `h = CreateFileW(ADDRESS, ...)`, doing a bit of work that takes a little time, and then calling `CloseHandle(h)`.

`os.stat()` won't be relevant to the error at all if Microsoft provides a better way to implement `os.stat()` that doesn't require opening a handle to the pipe, or if we decide that we should just inspect a "\\\\.\\PIPE\\" path as a string in `os.stat()` instead of opening a handle.

Reply:
Then how would you fix the test? os.close(os.open()) doesn’t fail without the fix.

Reply:
Maybe `os.open()` actually isn't fast enough to trigger the race condition reliably. Try `_winapi.CloseHandle(_overlapped.ConnectPipe(ADDRESS))`.

Reply:
I forgot that `os.stat()` also calls `GetFileAttributesW()` on the pipe while the handle is open. This tries to open a second pipe connection before the first one has been closed. The following reproduces the scenario:

```python
            h = _overlapped.ConnectPipe(ADDRESS)
            try:
                _winapi.CloseHandle(_overlapped.ConnectPipe(ADDRESS))
            except OSError as e:
                if e.winerror != _overlapped.ERROR_PIPE_BUSY:
                    raise
            finally:
                _winapi.CloseHandle(h)
 ```

In the test, 2/5 attempts at opening a second pipe connection fail with `ERROR_PIPE_BUSY`, so this error is intentionally ignored.

Reply:
Maybe we don’t need a test. All this is irrelevant for the fix. I can either delete the test, keep the PR as is, or close it unresolved. (Someone else can try again.)

Reply:
The code in my last comment reliably triggers the problem for me because it ensures that the client connects and disconnects from the server's next pipe instance, leaving the pipe in a broken state before the server calls `ConnectNamedPipe()`.

If it comes to closing the PR or deleting the test, I vote for deleting the test. It should have been designed from the beginning to handle the broken pipe error. Not every aspect of a correct design has to be tested.